shared/domains/okTurtles/events/events.test.js

```diff
@@ -0,0 +1,36 @@
+/* eslint-env mocha */
+
+import sinon from 'sinon'
+import { default as DATA } from '../data'
+import { default as EVENTS } from './index.js'
+import sbp from '../../../sbp'
+
+require('should-sinon')
+
+describe('[SBP] EVENTS domain', () => {
+  before(() => {
+    // TODO move this to events to make dependency more explicit
+    sbp.registerDomain('okTurtles.data', DATA)
+  })
+  it('should register event listener', () => {
+    const testListener = sinon.spy()
+    EVENTS['/on']('testEvent', testListener)
+    EVENTS['/emit']('testEvent')
+    EVENTS['/emit']('testEvent')
+    testListener.should.be.calledTwice()
+  })
+  it('should pass event listener the right data', () => {
+    const testListener = sinon.spy()
+    const testData = 1
+    EVENTS['/on']('testEvent', testListener)
+    EVENTS['/emit']('testEvent', testData)
+    testListener.should.be.calledWith({event: 'testEvent', data: testData})
+  })
+  it('should call "once" listener once', () => {
+    const testListener = sinon.spy()
+    EVENTS['/once']('testEvent', testListener)
+    EVENTS['/emit']('testEvent')
+    EVENTS['/emit']('testEvent')
+    testListener.should.be.calledOnce()
+  })
+})
```

Comment:
I'm thinking maybe it might be better for the SBP tests to run the functions through `sbp` so that they look more like how the code is used?

Also reminds me of:

- http://businessincubatornetworkinggroup.com/tag/unit-test-vs-integration-meme/
- https://me.me/i/2a-unit-testing-vs-integration-testing-12771711
- https://twitter.com/sgrif/status/952709272848093185

etc :)

Reply:
i.e. so that instead of `EVENTS['/once']` being tested, it tests the corresponding `sbp( .. )` call

Reply:
we already have integration tests - all the other tests we have :) - that prevent all those funny situations. (they also usually make it quite hard to debug them, as they just time out silently when failing. that's one of the reasons why it helps to also have unit-, or other more granular tests)

but sure, I can add SBP to these tests, and turn them into two-unit tests :) testing the domain implementations and the SBP functionality together. I am a bit worried to do so, let me try to explain why:   
writing unit tests for these small units is easy - anyone with some JS knowledge can do it and understand what's going on at the first glance. if we add SBP to the tests, we increase the "WTF per minute" level of the code, thus making it a step harder for potential volunteers to contribute, and for later contributors to maintain it. and we do so without much actual benefits - as the rest of the tests already check the integration with the SBP system.

Reply:
> if we add SBP to the tests, we increase the "WTF per minute" level of the code

But let me challenge you on that assumption: if we stick to just using `sbp` style, that is just one thing to learn and explain, whereas if suddenly there are two ways to call selectors, that's two different call styles that need to be learned, and this new one is never used anywhere else in the actual app code, only the tests. That was my reasoning at least *shrug*

Reply:
this _new one_ is just how you call a method in javascript, there is nothing special about it, and nothing new to learn if you know javascript. whereas when calling the sbp function instead, that is hiding away complexity that is not apparent for the first look, needs to be learned and explained.

(I am still planning to rewrite the tests based on your suggestion btw, I just wanted to bring in my take on easy-to-understand code)

Reply:
OK, as per my comment on Slack, I'm realizing that there is one issue you'll run into with modifying this to use `sbp` calls, and that's that it can be a pain to have to figure out how many `../`'s to put into the `import` call to import `sbp.js`.

So, since we're not yet turning this into an NPM module, I think there's another way that you can write `import "sbp"` at the top of the test files, which is to modify `.Gruntfile.babel.js` to add this line below line 267:

```
pathmodify.mod.id('sbp', p`${__dirname}/shared/sbp.js`)
```

I think from that point on you should be able to just write `import "sbp"` and for it to work.

Reply:
*err, per the Slack discussion, nevermind because the tests aren't run through browserify. So sorry I requested this change, you can just keep it as it is for now (at least until we think of something better or `sbp` gets published to NPM and/or we invent our own package manager :P)